qiskit/transpiler/coupling.py

```diff
@@ -428,6 +447,10 @@ def get_component_subgraphs(self, respect_direction: bool = False) -> List["Coup
             new_cmap = CouplingMap()
             new_cmap.graph = self.graph.subgraph(list(sorted(component)))
             output_list.append(new_cmap)
+        if respect_direction:
+            self._strong_components = output_list
+        else:
+            self._weak_components = output_list
         return output_list
```

Comment:
On second thoughts - both `list` and `CouplingMap` are mutable types.  If we cache, we'll need to be copying both of those on output.  Sorry to make a mess, but maybe it's better if we don't cache, and we can revisit it in the future if it _is_ a performance bottleneck.

Reply:
Heh ok, no worries, I'll drop the caching.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/9710/commits/4d293ba33276c80f2cacbc2818388d7568e143bd